libparsec/crates/client/src/workspace/merge.rs

```diff
@@ -263,27 +236,16 @@ pub(super) fn merge_local_folder_manifest(
     // /!\ Extra attention should be paid here if we want to add new fields
     // /!\ with their own sync logic, as this optimization may shadow them!
 
-    let merged_need_sync = merged_children != remote.children || merged_parent != remote.parent;
-    let updated = if merged_need_sync {
-        timestamp
-    } else {
-        remote.updated
-    };
-    // TODO: prevent sync pattern is not handled here !
-    let merged_manifest = LocalFolderManifest {
-        need_sync: merged_need_sync,
-        parent: merged_parent,
-        children: merged_children,
-        updated,
-        base: remote,
-        // TODO: confinement points support not implemented yet !
-        local_confinement_points: HashSet::new(),
-        // TODO: confinement points support not implemented yet !
-        remote_confinement_points: HashSet::new(),
-        speculative: false,
+    local_from_remote.speculative = false;
```

Comment:
I'd rather stick with the `merged need_sync` field: the fact we re-cycle `local_from_remote` to create `merged_manifest` instead of building a `LocalFolderManifest` from scratch is an implementation detail, but it makes the code more complex to read.

So I'd rather keep this behavior in a single place:
```rust
    let merged_need_sync = merged_children != remote.children || merged_parent != remote.parent;
    let merged_updated = if merged_need_sync {
        timestamp
    } else {
        remote.updated
    };
    let merged_base = local_from_remote.base;
    // TODO: Add a comment here to explain why we got the local confinement points from what seems to be the remote data ! 
    let merged_local_confinement_points = local_from_remote.local_confinement_points:,
    let merged_remote_confinement_points = local_from_remote.remote_confinement_points:,
    let merged_manifest = LocalFolderManifest {
       need_sync: merged_need_sync,
        parent: merged_parent,
        children: merged_children,
        updated: merged_updated,
        base: merged_base,
        local_confinement_points: merged_local_confinement_points:,
        remote_confinement_points: merged_remote_confinement_points:,
        speculative: false,
    };
```